app/features/project/project_controller.rb

```diff
@@ -70,77 +72,87 @@ class ProjectController < AuthenticatedControllerBase
+                                    provider_credential: provider_credential,
+                                    async_start: false)
+
+
+      fastfile = FastlaneCI::FastfilePeeker.peek(
+        git_repo: repo,
+        branch: branch
       )
 
-      # TODO: This should be refactored in some kind of FastlaneUtils` class.`
-      # We have synchronously cloned the repo, now we need to get the lanes.
-      repo_path = repo.local_folder
-      # First assume the fastlane directory and its file is in the root of the project
-      fastfiles = Dir[File.join(repo_path, "fastlane/Fastfile")]
-      # If not, it might be in a subfolder
-      fastfiles = Dir[File.join(repo_path, "**/fastlane/Fastfile")] if fastfiles.count == 0
-
-      if fastfiles.count > 1
-        logger.error("Ugh, multiple Fastfiles found, we're gonna have to build a selection in the future")
-        # for now, just take the first one
+      fastfile_config = {}
+
+      fastfile.tree.each_key do |key|
+        if key.nil?
+          fastfile_config[:no_platform] = fastfile.tree[key]
+        else
+          fastfile_config[key.to_sym] = fastfile.tree[key]
+        end
       end
 
-      fastfile_path = fastfiles.first
+      fastfile_config.to_json
+    end
+
+    get "#{HOME}/add/*/*" do |org, repo_name|
+      provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
 
-      parser = Fastlane::FastfileParser.new(path: fastfile_path)
-      available_lanes = parser.available_lanes
+      github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
+      selected_repo = github_service.repos.detect { |repo| org + "/" + repo_name == repo.full_name }
+
+      # We need to check whether we can checkout the project without issues.
+      # So a new project is created with default settings so we can fetch it.
+      repo_config = GitRepoConfig.from_octokit_repo!(repo: selected_repo)
 
       locals = {
           title: "Add new project",
-          repo: repo,
-          lanes: available_lanes,
-          fastfile_path: fastfile_path
+          repo: repo_config.full_name,
+          branches: github_service.branch_names(repo: repo_config.full_name)
       }
 
-      # Delete the project
-      FileUtils.rm_rf(repo_path) if File.directory?(File.join(repo_path, ".git"))
-
       erb(:new_project_form, locals: locals, layout: FastlaneCI.default_layout)
     end
 
-    post "#{HOME}/add/*" do |repo_name|
+    post "#{HOME}/add/*/*" do |org, repo_name|
       provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
 
       github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
-      selected_repo = github_service.repos.detect { |repo| repo_name == repo.name }
+      selected_repo = github_service.repos.detect { |repo| org + "/" + repo_name == repo.full_name }
```

Comment:
Instead of combining the string and then comparing the result, can we compare the repo name, and the org name directly?

Reply:
Have to check if it has that information but yeah😁